src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Schemas/OrderSchema.cs

```diff
@@ -194,18 +174,8 @@ public void Build(ISchema schema)
 
         private async Task<object> ResolveOrdersConnectionAsync(IMediator mediator, IResolveConnectionContext<object> context)
         {
-            var first = context.First;
-            var skip = Convert.ToInt32(context.After ?? 0.ToString());
-
-            var query = new SearchOrderQuery
-            {
-                Skip = skip,
-                Take = first ?? context.PageSize ?? 10,
-                Filter = context.GetArgument<string>("filter"),
-                Sort = context.GetArgument<string>("sort"),
-                CultureName = context.GetArgument<string>(nameof(Currency.CultureName).ToCamelCase()),
-                CustomerId = context.GetArgumentOrValue<string>("userId")
-            };
+            var query = AbstractTypeFactory<SearchOrderQuery>.TryCreateInstance();
+            query.Map(context);
```

Comment:
Create extension. Consider moving Line 180 inside thisextension.
```suggestion
            var query = context.ExtractQuery<SearchOrderQuery>();
```